src/java.base/share/classes/java/util/concurrent/CompletableFuture.java

```diff
@@ -407,7 +407,7 @@ private static Object reportJoin(Object r) {
             if ((x = ((AltResult)r).ex) == null)
                 return null;
             if (x instanceof CancellationException)
-                throw (CancellationException)x;
+                throw new CancellationException("", (CancellationException)x);
```

Comment:
One option here would be to put "CompletableFuture.join()" or "join()" as a message.